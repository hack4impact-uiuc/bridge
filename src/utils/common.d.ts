declare module "@hack4impact-uiuc/bridge/common" {
  import { Props, ReactType } from "react";
  import {
    ColorProps,
    SpaceProps,
    LayoutProps,
    TypographyProps as StyledTypographyProps,
    BordersProps as StyledBordersProps,
    BoxShadowProps,
  } from "styled-system";

  export interface BaseProps extends Props<any> {
    as?: ReactType;
    className?: string;
  }
  export interface CommonProps
    extends BaseProps,
      ColorProps,
      SpaceProps,
      LayoutProps {}

  export interface TypographyProps extends BaseProps, StyledTypographyProps {}

  export interface BorderProps extends StyledBordersProps, BoxShadowProps {}
}
