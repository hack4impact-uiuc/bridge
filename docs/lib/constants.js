export const COMPONENTS_URL = '/components';
export const DESIGN_URL = '/design';
