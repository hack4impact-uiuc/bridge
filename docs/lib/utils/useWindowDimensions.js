import { useState, useEffect } from 'react';

function getWindowDimensions() {
  const { innerWidth: width, innerHeight: height } = window;
  return {
    width,
    height,
  };
}

export default function useWindowDimensions() {
  const [windowDimensions, setWindowDimensions] = useState(getWindowDimensions());

  useEffect(() => {
    function handleResize() {
      const { innerWidth: width, innerHeight: height } = window;
      setWindowDimensions({ width, height });
    }

    window.addEventListener('resize', handleResize);
    return () => window.removeEventListener('resize', handleResize);
  }, []);

  return windowDimensions;
}
