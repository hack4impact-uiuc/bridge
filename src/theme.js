/* eslint-disable prefer-destructuring */

function fontStack(fonts) {
  return fonts.map((font) => (font.includes(' ') ? `"${font}"` : font)).join(', ');
}

const DEFAULT_SHADOW = '0 1px 6px 2px rgba(0,0,0,0.15)';

const baseColors = {
  hackBlue: ['#B4D8FA', '#3E87CD', '#155DA1', '#0E4E8A', '#0D3F6E', '#031425'],
  indigoBlue: ['#BCD0F3', '#415F94', '#2D4979', '#253E68', '#13294E', '#061734'],
  impactGrey: ['#FAFBFC', '#EBEEF2', '#8B9199', '#666B72', '#46494F'],
};

const hackBluePalette = {
  lighter: baseColors.hackBlue[0],
  light: baseColors.hackBlue[1],
  primary: baseColors.hackBlue[2],
  dark: baseColors.hackBlue[3],
  darker: baseColors.hackBlue[4],
  text: baseColors.hackBlue[5],
};

const indigoPalette = {
  lighter: baseColors.indigoBlue[0],
  light: baseColors.indigoBlue[1],
  primary: baseColors.indigoBlue[2],
  dark: baseColors.indigoBlue[3],
  darker: baseColors.indigoBlue[4],
  text: baseColors.indigoBlue[5],
};

const impactGreyPalette = {
  lighter: baseColors.impactGrey[0],
  light: baseColors.impactGrey[1],
  midtone: baseColors.impactGrey[2],
  primary: baseColors.impactGrey[3],
  darker: baseColors.impactGrey[4],
};

const secondaryColors = {
  green: ['#BDF3C6', '#90E19E', '#4C9859', '#256830', '#154C1E', '#031D07'],
  red: ['#FFBCBF', '#D1595F', '#A62B31', '#8A1A1F', '#6B1014', '#35080A'],
  yellow: ['#FFECBD', '#FFD15F', '#FDBE21', '#DA9C03', '#A77700', '#201803'],
};

const greenPalette = {
  lighter: secondaryColors.green[0],
  light: secondaryColors.green[1],
  primary: secondaryColors.green[2],
  dark: secondaryColors.green[3],
  darker: secondaryColors.green[4],
  text: secondaryColors.green[5],
};

const redPalette = {
  lighter: secondaryColors.red[0],
  light: secondaryColors.red[1],
  primary: secondaryColors.red[2],
  dark: secondaryColors.red[3],
  darker: secondaryColors.red[4],
  text: secondaryColors.red[5],
};

const yellowPalette = {
  lighter: secondaryColors.yellow[0],
  light: secondaryColors.yellow[1],
  primary: secondaryColors.yellow[2],
  dark: secondaryColors.yellow[3],
  darker: secondaryColors.yellow[4],
  text: secondaryColors.yellow[5],
};

const typographyColors = {
  charcoal: '#272A2D',
  iron: '#4B4E52',
  ink: '#0A162A',
  slate: '#363949',
};

const colors = {
  black: '#000000',
  white: '#FFFFFF',
  ...baseColors,
  ...secondaryColors,
  ...typographyColors,
  hackBluePalette,
  indigoPalette,
  impactGreyPalette,
  greenPalette,
  redPalette,
  yellowPalette,
  text: {
    grayScale: {
      main: typographyColors.charcoal,
      support: typographyColors.iron,
    },
    coolTone: {
      main: typographyColors.ink,
      support: typographyColors.slate,
    },
  },
  // states or in other words color variants
  variants: {
    primary: hackBluePalette,
    success: greenPalette,
    error: redPalette,
    warning: yellowPalette,
    secondary: hackBluePalette,
    white: {
      primary: '#FFFFFF',
      dark: '#FFFFFF',
      lighter: indigoPalette.light,
    },
  },
};

const fontSizes = ['12px', '14px', '16px', '18px', '20px', '22px', '25px', '28px', '32px', '45px', '60px'];

const fonts = {
  chivo: fontStack(['Chivo']),
  hkGrotesk: fontStack(['HKGrotesk']),
};

const fontWeights = {
  light: 300,
  normal: 400,
  medium: 500,
  bold: 700,
};

// styling specific to the Button component
const button = {
  font: fonts.chivo,
  fontWeight: fontWeights.bold,
  fontSize: '14px',
  letterSpacing: '2px',
  // default values, based on the color palette
  default: (colorPalette) => ({
    bg: {
      default: colorPalette.primary,
      hover: colorPalette.dark,
      disabled: colorPalette.lighter,
    },
    border: {
      default: 'none',
      hover: 'none',
      disabled: 'none',
    },
    fontColor: {
      default: colors.white,
      hover: colors.white,
      disabled: colorPalette.light,
    },
    shadow: {
      default: 'none',
      hover: 'none',
      disabled: 'none',
    },
    outline: {
      bg: {
        default: colors.white,
        hover: colors.white,
        disabled: colors.white,
      },
      border: {
        default: colorPalette.primary,
        hover: colorPalette.dark,
        disabled: colorPalette.lighter,
      },
      fontColor: {
        default: colorPalette.primary,
        hover: colorPalette.dark,
        disabled: colorPalette.lighter,
      },
      shadow: {
        default: 'none',
        hover: 'none',
        disabled: 'none',
      },
    },
  }),

  // everything below overrides the default values
  primary: {
    fontColor: {
      default: colors.white,
      hover: colors.white,
      disabled: colors.white,
    },
  },
  white: {
    fontColor: {
      default: indigoPalette.primary,
      hover: indigoPalette.primary,
      disabled: indigoPalette.lighter,
    },
    shadow: {
      hover: DEFAULT_SHADOW,
    },
    outline: {
      bg: {
        default: indigoPalette.primary,
        hover: indigoPalette.primary,
        disabled: indigoPalette.primary,
      },
      border: {
        disabled: indigoPalette.light,
      },
      fontColor: {
        default: colors.white,
        hover: colors.white,
        disabled: indigoPalette.lighter,
      },
      shadow: {
        hover: DEFAULT_SHADOW,
      },
    },
  },
  secondary: {
    bg: {
      hover: hackBluePalette.darker,
      disabled: impactGreyPalette.light,
    },
    fontColor: {
      disabled: impactGreyPalette.midtone,
    },
    outline: {
      bg: {
        disabled: colors.white,
      },
      border: {
        hover: hackBluePalette.darker,
        disabled: impactGreyPalette.light,
      },
      fontColor: {
        disabled: indigoPalette.lighter,
      },
    },
  },
};


const link = {
  default: (colorPalette) => ({
    color: {
      default: colorPalette.primary,
      hover: colorPalette.dark,
    },
  }),
  secondary: {
    default: hackBluePalette.primary,
    hover: hackBluePalette.darker,
  },
};


const theme = {
  // General
  colors,
  fonts,
  fontWeights,
  fontSizes,
  button,
  link,
  DEFAULT_SHADOW,
};


export default theme;
export { colors };
