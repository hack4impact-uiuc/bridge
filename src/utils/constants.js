import * as styledSystem from 'styled-system';
import systemPropTypes from '@styled-system/prop-types';
import PropTypes from 'prop-types';


const { compose, system } = styledSystem;

// Text Decoration Prop
export const textDecoration = system({
  prop: 'textDecoration',
  cssProperty: 'text-decoration',
});
textDecoration.propTypes = PropTypes.string;


// common style props to be passed into components
export const COMMON = compose(styledSystem.space, styledSystem.layout);
COMMON.propTypes = {
  ...systemPropTypes.space,
  ...systemPropTypes.layout,
};

// for easier access
export const TYPOGRAPHY = styledSystem.typography;
export const LAYOUT = styledSystem.layout;
export const POSITION = styledSystem.position;
export const FLEX = styledSystem.flexbox;
export const GRID = styledSystem.grid;
export const COLOR = styledSystem.color;
export const BORDER = styledSystem.border;

TYPOGRAPHY.propTypes = systemPropTypes.typography;
LAYOUT.propTypes = systemPropTypes.layout;
POSITION.propTypes = systemPropTypes.position;
FLEX.propTypes = systemPropTypes.flexbox;
GRID.propTypes = systemPropTypes.grid;
COLOR.propTypes = systemPropTypes.color;
BORDER.propTypes = systemPropTypes.border;
