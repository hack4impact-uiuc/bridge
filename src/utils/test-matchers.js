/* eslint-disable react/forbid-foreign-prop-types */
// taken from https://github.com/primer/components/blob/master/src/utils/test-matchers.js
import 'jest-styled-components';

const stringify = (d) => JSON.stringify(d, null, '  ');
/**
 * These are props that styled-system aliases for backwards compatibility.
 * For some reason, they don't show up in our toImplementSystemProps() matcher,
 * so we skip over them.
 */
const ALIAS_PROP_TYPES = ['w', 'align', 'justify', 'wrap'];

// eslint-disable-next-line no-undef
expect.extend({
  toImplementSystemProps(Component, propNames) {
    const propKeys = new Set(Object.keys(Component.propTypes));
    const expectedPropKeys = Object.keys(propNames.propTypes);
    const missing = expectedPropKeys.filter((key) => !propKeys.has(key))
      .filter((key) => !ALIAS_PROP_TYPES.includes(key));
    return {
      pass: missing.length === 0,
      message: () => `Missing prop${missing.length === 1 ? '' : 's'}: ${stringify(missing)}`,
    };
  },
});
