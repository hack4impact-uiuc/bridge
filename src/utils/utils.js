import themeGet from '@styled-system/theme-get';
import * as styledSystem from 'styled-system';
import theme from '../theme';

const { get: getKey } = styledSystem;

export const get = (key) => themeGet(key, getKey(theme, key));

export const getOrDefault = (val, def) => (val !== undefined && val !== null ? val : def);

export const lodashGet = (object, path, defaultVal) => {
  const PATH = Array.isArray(path)
    ? path
    : path.split('.').filter((i) => i.length);
  if (!PATH.length) {
    return object === undefined ? defaultVal : object;
  }
  if (object === null || object === undefined || typeof (object[PATH[0]]) === 'undefined') {
    return defaultVal;
  }
  return lodashGet(object[PATH.shift()], PATH, defaultVal);
};
