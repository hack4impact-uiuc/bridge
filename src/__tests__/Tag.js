import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { axe, toHaveNoViolations } from 'jest-axe';

import { Tag } from '..';
import theme, { colors } from '../theme';
import { COMMON } from '../utils/constants';
import { renderJSON } from '../utils/testing';
import 'babel-polyfill'; // axe violations required babel-polyfill

expect.extend(toHaveNoViolations);

describe('Tag', () => {
  it('renders a <div>', () => {
    expect(renderJSON(<Tag />).type).toEqual('div');
  });

  it('should have no axe violations', async () => {
    const { container } = render(<Tag>Blue</Tag>);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    cleanup();
  });

  it('implements system props', () => {
    expect(Tag).toImplementSystemProps(COMMON);
  });

  it('renders children', () => {
    const tagText = 'Blue';
    render(<Tag>{tagText}</Tag>);
    expect(screen.getByText(tagText)).toBeInTheDocument();
  });

  it('respects "height" props', () => {
    const { container } = render(<Tag height={200} />);
    expect(container.firstChild).toHaveStyle('height: 200px;');
  });

  it('renders typography using body styles', () => {
    const { container } = render(<Tag>Blue</Tag>);
    expect(container.firstChild).toHaveStyle(`font-family: ${theme.typography.body.fontFamily}`);
    expect(container.firstChild).toHaveStyle(`font-weight: ${theme.typography.body.fontWeight}px`);
    expect(container.firstChild).toHaveStyle(`font-size: ${theme.typography.body.fontSize}`);
    expect(container.firstChild).toHaveStyle(`letter-spacing: ${theme.typography.body.letterSpacing}`);
    expect(container.firstChild).toHaveStyle(`line-height: ${theme.typography.body.lineHeight}`);
  });

  it('renders blue primary color as default', () => {
    const { container } = render(<Tag>Blue</Tag>);
    expect(container.firstChild).toHaveStyle(`background: ${colors.bluePalette.primary}`);
    expect(container.firstChild).toHaveStyle(`color: ${colors.white}`);
  });

  it('renders dark text for light yellow tag', () => {
    const { container } = render(<Tag variant="light" color="yellow">Light Yellow</Tag>);
    expect(container.firstChild).toHaveStyle(`background: ${colors.yellowPalette.light}`);
    expect(container.firstChild).toHaveStyle(`color: ${colors.yellowPalette.text}`);
  });

  it('renders dark text for light green tag', () => {
    const { container } = render(<Tag variant="light" color="green">Light Green</Tag>);
    expect(container.firstChild).toHaveStyle(`background: ${colors.greenPalette.light}`);
    expect(container.firstChild).toHaveStyle(`color: ${colors.greenPalette.text}`);
  });
});

// Colors should match colors in default theme (design system styles)
describe('Tag Colors', () => {
  it('renders red primary', () => {
    expect(renderJSON(<Tag color="red" variant="primary">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders yellow primary', () => {
    expect(renderJSON(<Tag color="yellow" variant="primary">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders green primary', () => {
    expect(renderJSON(<Tag color="green" variant="primary">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders blue primary', () => {
    expect(renderJSON(<Tag color="blue" variant="primary">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders indigo primary', () => {
    expect(renderJSON(<Tag color="indigo" variant="primary">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders grey primary', () => {
    expect(renderJSON(<Tag color="grey" variant="primary">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders red light', () => {
    expect(renderJSON(<Tag color="red" variant="light">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders yellow light', () => {
    expect(renderJSON(<Tag color="yellow" variant="light">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders green light', () => {
    expect(renderJSON(<Tag color="green" variant="light">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders blue light', () => {
    expect(renderJSON(<Tag color="blue" variant="light">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders indigo light', () => {
    expect(renderJSON(<Tag color="indigo" variant="light">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders grey light', () => {
    expect(renderJSON(<Tag color="grey" variant="light">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders red dark', () => {
    expect(renderJSON(<Tag color="red" variant="dark">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders yellow dark', () => {
    expect(renderJSON(<Tag color="yellow" variant="dark">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders green dark', () => {
    expect(renderJSON(<Tag color="green" variant="dark">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders blue dark', () => {
    expect(renderJSON(<Tag color="blue" variant="dark">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders indigo dark', () => {
    expect(renderJSON(<Tag color="indigo" variant="dark">Tag</Tag>)).toMatchSnapshot();
  });
  it('renders grey dark', () => {
    expect(renderJSON(<Tag color="grey" variant="dark">Tag</Tag>)).toMatchSnapshot();
  });
});
