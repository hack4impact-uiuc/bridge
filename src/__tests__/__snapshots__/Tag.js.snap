// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Colors renders blue dark 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #0E4E8A;
  background-color: #0E4E8A;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders blue light 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #3E87CD;
  background-color: #3E87CD;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders blue primary 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #155DA1;
  background-color: #155DA1;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders green dark 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #256830;
  background-color: #256830;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders green light 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #031D07;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #90E19E;
  background-color: #90E19E;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders green primary 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #4C9859;
  background-color: #4C9859;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders grey dark 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #155DA1;
  background-color: #155DA1;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders grey light 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #EBEEF2;
  background-color: #EBEEF2;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders grey primary 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #666B72;
  background-color: #666B72;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders indigo dark 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #253E68;
  background-color: #253E68;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders indigo light 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #415F94;
  background-color: #415F94;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders indigo primary 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #2D4979;
  background-color: #2D4979;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders red dark 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #8A1A1F;
  background-color: #8A1A1F;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders red light 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #D1595F;
  background-color: #D1595F;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders red primary 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #A62B31;
  background-color: #A62B31;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders yellow dark 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #DA9C03;
  background-color: #DA9C03;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders yellow light 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #201803;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #FFD15F;
  background-color: #FFD15F;
}

<div
  className="c0"
>
  Tag
</div>
`;

exports[`Tag Colors renders yellow primary 1`] = `
.c0 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 16px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 24px;
  color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 8px;
  height: 28px;
  background: #FDBE21;
  background-color: #FDBE21;
}

<div
  className="c0"
>
  Tag
</div>
`;
